providers/azure/azure_provider.py

```diff
@@ -11,82 +11,186 @@
 
 
 class Azure_Provider:
-    def __init__(self):
+    def __init__(
+        self,
+        az_cli_auth: bool,
+        sp_env_auth: bool,
+        browser_auth: bool,
+        managed_entity_auth: bool,
+        subscription_ids: list,
+    ):
         logger.info("Instantiating Azure Provider ...")
-        self.credentials = DefaultAzureCredential()
+        self.credentials = self.__set_credentials__(
+            az_cli_auth, sp_env_auth, browser_auth, managed_entity_auth
+        )
+        self.identity = self.__set_identity_info__(
+            self.credentials,
+            az_cli_auth,
+            sp_env_auth,
+            browser_auth,
+            managed_entity_auth,
+            subscription_ids,
+        )
 
-    def get_credentials(self):
-        return self.credentials
+    def __set_credentials__(
+        self, az_cli_auth, sp_env_auth, browser_auth, managed_entity_auth
+    ):
+        # Browser auth creds cannot be set with DefaultAzureCredentials()
+        if not browser_auth:
+            if sp_env_auth:
+                self.__check_sp_creds_env_vars__()
+            try:
+                # Since the input vars come as True when it is wanted to be used, we need to inverse it since
+                # DefaultAzureCredential sets the auth method excluding the others
+                credentials = DefaultAzureCredential(
+                    exclude_environment_credential=not sp_env_auth,
+                    exclude_cli_credential=not az_cli_auth,
+                    exclude_managed_identity_credential=not managed_entity_auth,
+                    exclude_visual_studio_code_credential=True,
+                    exclude_shared_token_cache_credential=True,
+                    exclude_powershell_credential=True,
+                )
+            except Exception as error:
+                logger.critical("Failed to retrieve azure credentials")
+                logger.critical(
+                    f"{error.__class__.__name__}[{error.__traceback__.tb_lineno}] -- {error}"
+                )
+        else:
+            credentials = InteractiveBrowserCredential()
 
+        return credentials
 
-def check_credential_env_vars() -> Azure_Identity_Info:
-    for env_var in ["AZURE_CLIENT_ID", "AZURE_TENANT_ID", "AZURE_CLIENT_SECRET"]:
-        if not getenv(env_var):
+    def __check_sp_creds_env_vars__(self):
```

Comment:
I prefer this name since the only way to check the service principal credentials is using environment variables.

```suggestion
    def __check_service_principal_credentials(self):
```

Reply:
We are actually getting the env variables that includes service principal information, we are not checking credentials